File: Classwork/mobileApp/src/anonymousClassesDemos/Demo4.java
package Classwork.mobileApp.src.anonymousClassesDemos;

import javax.swing.*;
import java.awt.*;
import java.awt.event.MouseEvent;
import java.awt.event.MouseListener;

import static javax.swing.WindowConstants.EXIT_ON_CLOSE;

public class Demo4 extends JPanel {
	public Demo4() {
		var frame = new JFrame();
		frame.add(this);
		frame.setVisible(false);
		frame.setSize(new Dimension(Toolkit.getDefaultToolkit().getScreenSize()));
		setFocusable(true);
		addMouseListener(new MouseListener() {
			@Override
			public void mouseClicked(MouseEvent e) {

			}

			@Override
			public void mousePressed(MouseEvent e) {

			}

			@Override
			public void mouseReleased(MouseEvent e) {

			}

			@Override
			public void mouseEntered(MouseEvent e) {
				while (true) System.out.println("Mouse Coordinates: (" + e.getX() + ", " + e.getY() + ')');
			}

			@Override
			public void mouseExited(MouseEvent e) {

			}
		});
		frame.setDefaultCloseOperation(EXIT_ON_CLOSE);
	}

	public static void main(String[] args) {
		new Demo4();
	}
}

File: Classwork/mobileApp/src/anonymousClassesDemos/Demo4.kt
package anonymousClassesDemos

import java.awt.Dimension
import java.awt.Toolkit
import java.awt.event.MouseEvent
import java.awt.event.MouseListener
import javax.swing.JFrame
import javax.swing.JPanel
import javax.swing.WindowConstants.EXIT_ON_CLOSE

class Demo4 : JPanel() {
	companion object {
		@JvmStatic
		fun main(args: Array<String>) {
			Demo4()
		}
	}

	init {
		val frame = JFrame()
		frame.add(this)
		frame.isVisible = false
		frame.size = Dimension(Toolkit.getDefaultToolkit().screenSize)
		isFocusable = true
		addMouseListener(object : MouseListener {
			override fun mouseClicked(e: MouseEvent) {}
			override fun mousePressed(e: MouseEvent) {}
			override fun mouseReleased(e: MouseEvent) {}
			override fun mouseEntered(e: MouseEvent) {
				while (true) println("Mouse Coordinates: (" + e.x + ", " + e.y + ')')
			}

			override fun mouseExited(e: MouseEvent) {}
		})
		frame.defaultCloseOperation = EXIT_ON_CLOSE
	}
}